presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchRecordCursor.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.elasticsearch;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.RecordCursor;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.json.JsonCodec;
+import io.airlift.slice.Slice;
+import io.airlift.units.Duration;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.search.SearchHit;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.facebook.presto.elasticsearch.ElasticsearchErrorCode.ELASTIC_SEARCH_MAX_HITS_EXCEEDED;
+import static com.facebook.presto.elasticsearch.ElasticsearchUtils.serializeObject;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static io.airlift.slice.Slices.EMPTY_SLICE;
+import static io.airlift.slice.Slices.utf8Slice;
+import static java.util.Objects.requireNonNull;
+
+public class ElasticsearchRecordCursor
+        implements RecordCursor
+{
+    private static final JsonCodec<Object> VALUE_CODEC = jsonCodec(Object.class);
+
+    private final List<ElasticsearchColumnHandle> columnHandles;
+    private final Map<String, Integer> jsonPathToIndex = new HashMap<>();
+    private final int maxHits;
+    private final Iterator<SearchHit> searchHits;
+    private final Duration timeout;
+    private final ElasticsearchQueryBuilder builder;
+
+    private long totalBytes;
+    private List<Object> fields;
+
+    public ElasticsearchRecordCursor(List<ElasticsearchColumnHandle> columnHandles, ElasticsearchConnectorConfig config, ElasticsearchSplit split)
+    {
+        this.columnHandles = requireNonNull(columnHandles, "columnHandle is null");
```

Comment:
The preferred clean way for `requireNonNull` checks (as I have been told previously in some reviews):

```
requireNonNull(columnHandles, "columnHandle is null");
this.columnHandles = columnHandles;
```

instead of 

```
this.columnHandles = requireNonNull(columnHandles, "columnHandle is null");
```